RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -675,26 +1098,56 @@ TauVars AntiElectronIDMVA6<TauType, ElectronType>::getTauVarsTypeSpecific(const
   tauVars.leadChargedPFCandEtaAtEcalEntrance = theTau.etaAtEcalEntranceLeadChargedCand();
   tauVars.leadChargedPFCandPt = theTau.ptLeadChargedCand();
   tauVars.phi = theTau.phi();
-  if (!usePhiAtEcalEntranceExtrapolation_) {
-    tauVars.phi = theTau.phiAtEcalEntrance();
+  if (!isPhase2_) {
+    if (!usePhiAtEcalEntranceExtrapolation_) {
+      tauVars.phi = theTau.phiAtEcalEntrance();
+    } else {
+      float sumPhiTimesEnergy = 0.;
+      float sumEnergy = 0.;
+      for (const auto& candidate : theTau.signalCands()) {
+        float phiAtECalEntrance = candidate->phi();
+        bool success = false;
+        reco::Candidate::Point posAtECal = positionAtECalEntrance_(candidate.get(), success);
+        if (success) {
+          phiAtECalEntrance = posAtECal.phi();
+        }
+        sumPhiTimesEnergy += phiAtECalEntrance * candidate->energy();
+        sumEnergy += candidate->energy();
+      }
+      if (sumEnergy > 0.) {
+        tauVars.phi = sumPhiTimesEnergy / sumEnergy;
+      }
+    }
+    tauVars.emFraction = std::max(theTau.emFraction_MVA(), 0.f);
   } else {
-    float sumPhiTimesEnergy = 0.;
-    float sumEnergy = 0.;
-    for (const auto& candidate : theTau.signalCands()) {
-      float phiAtECalEntrance = candidate->phi();
+    if (std::abs(theTau.eta()) >= ecalBarrelEndcapEtaBorder_) {  //HGCal
+      tauVars.etaAtEcalEntrance = -99.;
+      tauVars.leadChargedPFCandEtaAtEcalEntrance = -99.;
       bool success = false;
-      reco::Candidate::Point posAtECal = positionAtECalEntrance_(candidate.get(), success);
+      reco::Candidate::Point posAtECal = positionAtECalEntrance_(theTau.leadChargedHadrCand().get(), success);
       if (success) {
-        phiAtECalEntrance = posAtECal.phi();
+        tauVars.leadChargedPFCandEtaAtEcalEntrance = posAtECal.eta();
+      }
+      float sumEtaTimesEnergy = 0.;
+      float sumEnergy = 0.;
+      for (const auto& candidate : theTau.signalCands()) {
+        float etaAtECalEntrance = candidate->eta();
+        success = false;
+        posAtECal = positionAtECalEntrance_(candidate.get(), success);
+        if (success) {
+          etaAtECalEntrance = posAtECal.eta();
+        }
+        sumEtaTimesEnergy += etaAtECalEntrance * candidate->energy();
+        sumEnergy += candidate->energy();
+      }
```

Comment:
this repeats multiple times; please wrap into a function and reuse

Reply:
Done, but it repeats just 2 times. Other similar computations are within loops doing also other checks so including it there will either over-complicate the function or multiply loops over same collections.